src/pykeen/models/base.py

```diff
@@ -32,27 +33,253 @@
+            The indices of (head, tail) pairs.
+
+        :return: shape: (batch_size, num_relations), dtype: float
+            For each h-t pair, the scores for all possible relations.
+        """
+
+    @abstractmethod
+    def score_h(self, rt_batch: torch.LongTensor) -> torch.FloatTensor:
+        """Forward pass using left side (head) prediction.
+
+        This method calculates the score for all possible heads for each (relation, tail) pair.
+
+        :param rt_batch: shape: (batch_size, 2), dtype: long
+            The indices of (relation, tail) pairs.
+
+        :return: shape: (batch_size, num_entities), dtype: float
+            For each r-t pair, the scores for all possible heads.
+        """
+
+    """Abstract methods - loss computation"""
+
+    @abstractmethod
+    def compute_mr_loss(
+        self,
+        positive_scores: torch.FloatTensor,
+        negative_scores: torch.FloatTensor,
+    ) -> torch.FloatTensor:
+        """Compute the mean ranking loss for the positive and negative scores.
+
+        :param positive_scores:  shape: s, dtype: float
+            The scores for positive triples.
+        :param negative_scores: shape: s, dtype: float
+            The scores for negative triples.
+        :raises RuntimeError:
+            If the chosen loss function does not allow the calculation of margin ranking
+        :return: dtype: float, scalar
+            The margin ranking loss value.
+        """
+
+    @abstractmethod
+    def compute_label_loss(
+        self,
+        predictions: torch.FloatTensor,
+        labels: torch.FloatTensor,
+    ) -> torch.FloatTensor:
+        """Compute the classification loss.
+
+        :param predictions: shape: s
+            The tensor containing predictions.
+        :param labels: shape: s
+            The tensor containing labels.
+
+        :return: dtype: float, scalar
+            The label loss value.
+        """
+
+    @abstractmethod
+    def compute_self_adversarial_negative_sampling_loss(
+        self,
+        positive_scores: torch.FloatTensor,
+        negative_scores: torch.FloatTensor,
+    ) -> torch.FloatTensor:
+        """Compute self adversarial negative sampling loss.
+
+        :param positive_scores: shape: s
+            The tensor containing the positive scores.
+        :param negative_scores: shape: s
+            Tensor containing the negative scores.
+        :raises RuntimeError:
+            If the chosen loss does not allow the calculation of self adversarial negative sampling losses.
+        :return: dtype: float, scalar
+            The loss value.
+        """
+
+    """Concrete methods"""
+
+    def _set_device(self, device: DeviceHint = None) -> None:
+        """Set the Torch device to use."""
+        self.device = resolve_device(device=device)
+
+    def to_cpu_(self) -> Model:
+        """Transfer the entire model to CPU."""
+        self._set_device('cpu')
+        return self.to_device_()
+
+    def to_gpu_(self) -> Model:
```

Comment:
Do we need these methods? Since it subclasses from `nn.Module` it has already `cpu()` and `cuda()`. I think these methods are a relic from the time where we manually moved regularization terms around. The better solution is to properly register such as buffers, using `register_buffer`.

Reply:
We do not need `to_gpu_()`. It is never used. `to_cpu_()` is only used once, and so it should be replaced

Reply:
I removed these in 87660e47